packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -210,4 +217,70 @@ export class MetadataOrchestrator {
         return 'string';
     }
   };
+
+  public eligibilityDelegate = async (
+    requests: ApexOASEligiblePayload,
+    maxRetries: number = 3
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    let response = {};
+    let attempt = 0;
+    const languageClient = languageClientUtils.getClientInstance();
+    if (languageClient) {
+      while (attempt < maxRetries) {
```

Comment:
I am still not convinced that retries are necessary.

Reply:
Yes I still feel that way. Let me remove it first and see how it will work.